test/tools/save-logs/save_logs.py

```diff
@@ -0,0 +1,385 @@
+    It is a best effort to remove, so if it fails, will generate an message
+    to the user to clean up manually
+    :param testhost: ip address of stack
+    :param args: command line arguments
+    '''
+    remote_ssh_res = {'stdout': "", 'exitcode': 0}
+    str_remove_cmd = "rm /tmp/" + gather_script
+
+    remote_ssh_res = remote_shell(str_remove_cmd, args)
+    if remote_ssh_res['exitcode'] != 0:
+        print("Error in remove command from " + testhost)
+        print("Please clean up /tmp directory on stack")
+
+
+def run_gather_logs(testhost, args):
+    '''
+    This function will run the rackhd_gather_info.py script on the stack via remote ssh.
+    If the script runs successfully, the stdout of the script produces a message that
+    we check for that includes the package name, in this form:
+    Package: /tmp/rackhd/pkg/RackHDLogs-ora-20150715-232238.tgz
+    :param testhost: ip address of rackhd server
+    :param args: command line arguments
+    :return:
+        none as the package name on any failure
+        package name in the form of /tmp/rackhd/.....tgz on success
+    '''
+
+    pkgfile = "none"
+    remote_ssh_res = {'stdout': "", 'exitcode': 0}
+    str_gather_cmd = "/tmp/" + gather_script
+
+    remote_ssh_res = remote_shell(str_gather_cmd, args)
+    if remote_ssh_res['exitcode'] != 0:
+        print("Error in executing " + gather_script + " on " + testhost)
+    else:
+        pstr = remote_ssh_res['stdout']
+        # the rackhd_gather_info.py script returns the package name in the output
+        for row in pstr.split('\n'):
+            if 'Package:' in row:
+                pkgfile = row.split(': ')[1]
+    return pkgfile
+
+
+def scp_tgz_to_logserver(testhost, pkginfo, logdir, args):
+    '''
+    Using pexpect, this function will scp the tarfile specified in pkginfo from the
+    testhost stack to the log directory.  The pkginfo contains the full path on the stack.
+    :param testhost: ip address of rackhd server
+    :param pkginfo: string returned from gather script
+    :param logdir: full local path to bug directory
+    :param args: command line arguments
+    :return:
+        True if scpcommanddoes not return an error
+        False if an error in the scpcommandfailed
+    '''
+    status = False
+
+    command = "scp " + args.port2 + " " + args.user + "@" + testhost + ":" + pkginfo + " " + logdir
+    child = pexpect.spawn(command)
+    child.timeout = 60
+    i = child.expect(['assword: ', pexpect.TIMEOUT])
+    if i == 0:
+        child.sendline(args.pwd)
+        # wait for scp to copy file
+        time.sleep(10)
+        status = True
+    else:
+        print(child.before)
+        print("ERROR: Expecting password for SCP timed out.")
+        print("SCP of log files failed due to password error.")
+        print("Please manually scp " + pkginfo + " to desired location.")
+
+    child.close()
+    return status
+
+
+def handle_pexpect_error(child, errstr):
```

Comment:
This comment applies to the file extract_logs.py instead of here.  I removed the function from that file.